aten/src/ATen/native/Fill.cpp

```diff
@@ -10,7 +10,39 @@ namespace native {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ fill ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+namespace {
+  template <typename scalar_t>
+  inline void fill_fast(Tensor& self, Scalar value_scalar) {
+    auto value = value_scalar.to<scalar_t>();
+    scalar_t * dptr = reinterpret_cast<scalar_t *>(self.data_ptr());
+    *dptr = value;
```

Comment:
This only really works if the tensor is on CPU, right? Is `fill_out` only called for CPU tensors? Could we toss an assert in that the device is correct?

Reply:
Oh yea let me add the guard back.